diff --git a/.gitignore b/.gitignore
index 4105cfbd6f26..986174004119 100644
--- a/.gitignore
+++ b/.gitignore
@@ -54,23 +54,11 @@ Module.symvers
 /Module.markers
 
 #
-# Debian directory (make deb-pkg)
-#
-/debian/
-
-#
 # tar directory (make tar*-pkg)
 #
 /tar-install/
 
 #
-# git files that we don't want to ignore even if they are dot-files
-#
-!.gitignore
-!.mailmap
-!.cocciconfig
-
-#
 # Generated include files
 #
 include/config
@@ -116,5 +104,12 @@ all.config
 # Kdevelop4
 *.kdev4

+#
+# Debian packaging: ignore everything at the top level, since it isn't
+# included in our repository
+#
+/*
+!/debian/
+
 # fetched Android config fragments
 kernel/configs/android-*.cfg
